src/main/java/net/library/controller/UserController.java

```diff
@@ -48,5 +59,4 @@ public UserDto getUserById(@PathVariable(required = false, value = "user_id") fi
     public void deleteById(@PathVariable(required = false, value = "user_id") final UUID userId) {
         service.deleteById(userId);
     }
-}
-   
\ No newline at end of file
+}
```

Comment:
Find this `return UserMapper.toDto(service.getUserById(userId));` line of code and test it.
It is potential NPE